OrbitCore/Capture.cpp

```diff
@@ -35,21 +34,19 @@ void Capture::SetTargetProcess(std::shared_ptr<Process> process) {
   if (process != GTargetProcess) {
     GSamplingProfiler = std::make_shared<SamplingProfiler>(process);
     GTargetProcess = std::move(process);
-    GSelectedFunctionsMap.clear();
   }
 }
 
-ErrorMessageOr<void> Capture::StartCapture() {
+ErrorMessageOr<void> Capture::StartCapture(
+    uint64_t process_id, std::string process_name,
+    absl::flat_hash_map<uint64_t, FunctionInfo> selectedFunctions) {
   if (GTargetProcess->GetID() == 0) {
     return ErrorMessage(
         "No process selected. Please choose a target process for the capture.");
   }
 
-  capture_data_ =
-      CaptureData(GTargetProcess->GetID(), GTargetProcess->GetName(),
```

Comment:
Why are you removing the use of GTargetProcess and providing it as a parameter of StartCapture? If GTargetProcess is being deprecated I think it should be removed entirely then. I think it makes confusing the flow of how a process is targeted. And as result, SetTargetProcess should dissapear as well

Reply:
As a clarification/extension of the previous comment: If GTargetProcess is not being removed, I think this should not be updated then because as I understand it, it is just including a extra step: clients use SetGTargetProcess to set the target process, then the client gets the GTargetProcess data again and provides it to Capture, which has the GTargetProcess information already. I think that if we want to provide the target process when StartCapture is called directlly, GTargetProcess is duplicating unnecessary code so it should be removed and handled in the client entirely. If I am not understanding the flow please let me know.

Reply:
GTargetProcess will be removed as a next step. And as we have all information required here also available, I decided to already pass the information here.

Reply:
I think it should be done in the next step then, to have everything in a same PR

Reply:
I kind of reverted some of the changes. However, as I needed to change the constructor anyways and already have that information available on the caller side, I would prefer to keep this change here.

Reply:
Ok, I understand, thanks!